src/components/AllArticlesSortBy.jsx

```diff
@@ -0,0 +1,72 @@
+import { useState } from "react";
+import { Button } from "react-bootstrap";
+
+const AllArticlesSortBy = ({ setSearchParams }) => {
+  const [menuParams, setMenuParams] = useState({
+    sort_by: "",
+    order: "asc",
+  });
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setMenuParams({ ...menuParams, [name]: value });
+  };
+
+  const handleClick = () => {
+    const paramsString = `sort_by=${menuParams.sort_by}&order=${menuParams.order}`;
+    const searchParams = new URLSearchParams(paramsString);
+
+    if (menuParams.sort_by === "") {
```

Comment:
A very graceful way of getting around that issue we were discussing yesterday of what if the user tries to input only one of the values! Alternatively you could make the select input tags 'required' but I'm not sure off the top of my head how that looks with a select specifically.